tiledb/api/src/query/read/callback.rs

```diff
@@ -442,11 +442,15 @@ macro_rules! query_read_callback {
 
                         let [< l_ $U:snake >] = self.[< arg_ $U:snake >].location.borrow();
                         let [< input_ $U:snake >] = [< l_ $U:snake >].as_shared();
+                            /*
+                             * TODO: if it is the final result, enable moving out of
+                             * this so that we can avoid a copy
+                             */
 
                         let [< arg_ $U:snake >] = RawReadOutput {
                             nvalues: [< nvalues_ $U:snake >],
                             nbytes: [< nbytes_ $U:snake >],
-                            input: &[< input_ $U:snake >]
+                            input: [< input_ $U:snake >]
```

Comment:
Is it just me, or is paste just a fancy `##` symbol concatenation thing?

Reply:
Pretty much yep. Except normal rust macros don't let you do that.